### PR TITLE
tests and csv

### PR DESCRIPTION
**Pull request by andrewfraser73**
add CSV download to the report

### CODE REVIEW

File: src/Service.Host/client/src/components/reports/PrefSupReceiptsReport.js [hunk 1]
```diff
@@ -30,11 +31,20 @@ function PrefSupReceiptsReport() {
...
+
     return (
         <Page history={history} homeUrl={config.appRoot}>
             <Grid container spacing={3}>
                 <Grid item xs={12}>
                     <Title text="Pref Sup Receipts Report" />
+                    <ExportButton
+                        href={
+                            `${config.appRoot}/purchasing/reports/pref-sup-receipts/export` +
+                            `?fromDate=${options.fromDate.toISOString()}` +
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/reports/PrefSupReceiptsReport.js:44]: queryString.stringify(options) does this much cleaner

**Review by lewisrenfrew — APPROVED** [review]
Not sure why there's a huge package-lock diff but I assume it works ok. Remember you can do 

npm ci

instead of npm install

to install dependencies without changing the package-lockl

File: src/Service/Modules/Reports/PrefSupReceiptsReportModule.cs [hunk 2]
```diff
@@ -36,5 +38,17 @@ private async Task GetApp(HttpRequest req, HttpResponse res)
 
             await res.Negotiate(results);
         }
+
+        private async Task GetExport(
```
Comment (richardiphillips) [inline src/Service/Modules/Reports/PrefSupReceiptsReportModule.cs:42]: test?

File: src/Service/Modules/Reports/PrefSupReceiptsReportModule.cs [hunk 3]
```diff
@@ -17,12 +18,13 @@ public class PrefSupReceiptsReportModule : ICarterModule
...
         {
             app.MapGet("/purchasing/reports/pref-sup-receipts/report", this.GetReport);
+            app.MapGet("/purchasing/reports/pref-sup-receipts/export", this.GetExport);
             app.MapGet("/purchasing/reports/pref-sup-receipts", this.GetApp);
         }
 
         private async Task GetApp(HttpRequest req, HttpResponse res)
         {
-            await res.Negotiate(new ViewResponse { ViewName = "Index.html" });
+            await res.Negotiate(new ViewResponse {ViewName = "Index.html"});
```
Comment (richardiphillips) [inline src/Service/Modules/Reports/PrefSupReceiptsReportModule.cs:27]: stylecop seems to be off?

File: src/Service.Host/client/src/components/reports/PrefSupReceiptsReport.js [hunk 4]
```diff
@@ -30,11 +31,20 @@ function PrefSupReceiptsReport() {
 
     const reportData = useSelector(state => state[prefSupReceiptsReport.item]?.data);
 
+    console.log(options.fromDate.toISOString());
```
Comment (richardiphillips) [inline src/Service.Host/client/src/components/reports/PrefSupReceiptsReport.js:34]: should probably kill this before going live

**Review by richardiphillips — APPROVED** [review]

**Pull request merged**